src/lib/Lib/MrDocsCompilationDatabase.cpp

```diff
@@ -165,6 +167,40 @@ adjustCommandLine(
     return new_cmdline;
 }
 
+std::optional<std::string> 
+executeCmakeExportCompileCommands(llvm::StringRef cmakePath, llvm::StringRef cmakeListsPath) 
+{
+    if ( ! llvm::sys::fs::exists(cmakePath)) {
+        return std::nullopt;
+    }    
+    if ( ! llvm::sys::fs::exists(cmakeListsPath)) {
+        return std::nullopt;
+    }
+
+    llvm::SmallString<128> stdOutPath;
+    if (auto ec = llvm::sys::fs::createTemporaryFile("stdout", "txt", stdOutPath)) 
+    {
+        return std::nullopt;
+    }
+
+    llvm::SmallString<128> stdErrPath;    
+    if (auto ec = llvm::sys::fs::createTemporaryFile("stderr", "txt", stdErrPath)) 
+    {
+        return std::nullopt;
+    }
+
+    std::optional<llvm::StringRef> const redirects[] = {llvm::StringRef(), stdOutPath.str(), stdErrPath.str()};
+    std::vector<llvm::StringRef> const args = {cmakePath, cmakeListsPath, "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"};
```

Comment:
I thought `cmakeListsPath` was the `CMakeLists.txt` path. It's actually its parent. This could be `projectPath` or something then.

Reply:
Projects usually have their requirements for CMake configure args. Only `DCMAKE_EXPORT_COMPILE_COMMANDS=ON` rarely works as configure args. These need to come from mrdocs.yml.

Reply:
I'm not sure how cmake can run without the source and build paths as arguments.

And all the intermediary cmake build directories need to be in a temporary directory, where the compile_commands.json will also end up.

Reply:
If the default generator is Visual Studio, this needs to change to Ninja because Visual Studio can't generate compile_commands.json.